packages/python-solv/python-solv.spec

```
@@ -0,0 +1,160 @@
+
+
+
+%prep
+%autosetup -p1 -n lib%{libname}-%{version}
+
+%build
+%cmake -GNinja                                            \
+  -DFEDORA=1                                              \
+  -DENABLE_RPMDB=ON                                       \
+  -DENABLE_RPMDB_BYRPMHEADER=ON                           \
+  -DENABLE_RPMDB_LIBRPM=ON                                \
+  -DENABLE_RPMPKG_LIBRPM=ON                               \
+  -DENABLE_RPMMD=ON                                       \
+  -DENABLE_COMPS=%{__cmake_switch -b comps}               \
+  -DENABLE_APPDATA=%{__cmake_switch -b appdata}           \
+  -DUSE_VENDORDIRS=ON                                     \
+  -DWITH_LIBXML2=ON                                       \
+  -DWITH_OPENSSL=ON                                       \
+  -DENABLE_LZMA_COMPRESSION=ON                            \
+  -DENABLE_BZIP2_COMPRESSION=ON                           \
+  -DENABLE_ZSTD_COMPRESSION=%{__cmake_switch -b zstd}     \
+  -DENABLE_ZCHUNK_COMPRESSION=%{__cmake_switch -b zchunk} \
+%if %{with zchunk}
+  -DWITH_SYSTEM_ZCHUNK=ON                                 \
+%endif
+  -DENABLE_HELIXREPO=%{__cmake_switch -b helix_repo}      \
+  -DENABLE_SUSEREPO=%{__cmake_switch -b suse_repo}        \
+  -DENABLE_DEBIAN=%{__cmake_switch -b debian_repo}        \
+  -DENABLE_ARCHREPO=%{__cmake_switch -b arch_repo}        \
+  -DMULTI_SEMANTICS=%{__cmake_switch -b multi_semantics}  \
+  -DENABLE_COMPLEX_DEPS=%{__cmake_switch -b complex_deps} \
+  -DENABLE_CONDA=%{__cmake_switch -b conda}               \
+  -DENABLE_PYTHON=%{__cmake_switch -b python_bindings}    \
+  -DENABLE_STATIC=ON                                      \
+  -DDISABLE_SHARED=ON                                     \
+  -DPYTHON_LIBRARY=/usr/lib64/libpython%{python3_pkgversion}.so.1.0        \
+  -DPYTHON_INCLUDE_DIR=/usr/include/python%{python3_pkgversion}            \
+  -DPYTHON_EXECUTABLE=%{python3}                          \
+  %{nil}
+%cmake_build
+
+%install
+%cmake_install
+
+
+rm -rf %{buildroot}%{_includedir}/solv/
+
+rm %{buildroot}%{_bindir}/*solv
+rm %{buildroot}%{_bindir}/installcheck
+rm %{buildroot}%{_mandir}/man*/*solv*
+rm %{buildroot}%{_mandir}/man1/installcheck.1*
+rm %{buildroot}%{_datadir}/cmake/Modules/FindLibSolv.cmake
+rm %{buildroot}%{_libdir}/libsolv*.a
+rm %{buildroot}%{_libdir}/pkgconfig/libsolv*.pc
+
+
+%check
+%ctest
+
+# Python smoke test (not tested in %%ctest):
+export PYTHONPATH=%{buildroot}%{python3_sitearch}
+export LD_LIBRARY_PATH=%{buildroot}%{_libdir}
+%python3 -c 'import solv'
+
+%package -n     python%{python3_pkgversion}-%{libname}
+Summary:        %{summary}
+%{?python_provide:%python_provide python%{python3_pkgversion}-%{libname}}
+
+%description -n python%{python3_pkgversion}-%{libname}
+%{summary}
+
+
+%files -n python%{python3_pkgversion}-%{libname}
+%license LICENSE*
+%doc README
+%{python3_sitearch}/_%{libname}.so
+%{python3_sitearch}/%{libname}.py
+%{python3_sitearch}/__pycache__/%{libname}.*
```

https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#_files_to_include suggests this:
```suggestion
%pycached %{python3_sitelib}/%{libname}.py
```

Not 100% sure that also works on EL8.

---

It seems so, but we're not using this anywhere in our packaging, and I feel like solv is the wrong package to start experimenting ;)